api/tests/mocha/services/users.spec.js

```diff
@@ -1288,6 +1288,33 @@ describe('Users service', () => {
       });
     });
 
+    it('succeeds if only language is defined by a user with full access', () => {
+      const data = {
+        language: 'es'
+      };
+      service.__set__('validateUser', sinon.stub().resolves({}));
+      service.__set__('validateUserSettings', sinon.stub().resolves({}));
+      sinon.stub(db.medic, 'put').resolves({});
+      sinon.stub(db.users, 'put').resolves({});
+      return service.updateUser('paul', data, true).then(() => {
+        chai.expect(db.medic.put.callCount).to.equal(1);
```

Comment:
Re: the merge problem. 
I we had had assertions here about how the arguments to put looked like, and making sure it doesn't contain `language`, the unit test would have caught the merge issue.

Reply:
Good call. I have updated these tests to assert the arguments passed to medic/users.